File: common/src/main/java/com/github/fcopardo/easyrest/common/callbacks/AfterTaskFailure.java
package com.github.fcopardo.easyrest.common.callbacks;

/**
 * Implement this interface in the summoner class to perform any fallback operations when the rest call fails.
 * Created by Fco Pardo on 8/24/14.
 */
public interface AfterTaskFailure {
    <L extends Exception> void onTaskFailed(L e);
}

File: common-api/src/main/kotlin/com/github/pardo/easyrest/api/callbacks/AfterTaskFailure.kt
package com.github.fcopardo.easyrest.api.callbacks

/**
 * Implement this interface in the summoner class to perform any fallback operations when the rest call fails.
 * Created by Fco Pardo on 8/24/14.
 */
interface AfterTaskFailure {
    fun <L : Exception> onTaskFailed(e: L)
}
